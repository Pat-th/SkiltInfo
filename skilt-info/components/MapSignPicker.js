import React, { useEffect, useState } from 'react';
import { StyleSheet, Text, View, FlatList, Dimensions } from 'react-native';
import Modal from "react-native-modal";
import SignPreview from "../components/SignPreview";
import MapView, { Marker } from "react-native-maps";

const MapSignPicker = props => {
    const [navigation, setNavigation] = useState(null);
    const [picture, setPicture] = useState(null);
    const [latitude, setLatitude] = useState(null);
    const [longitude, setLongitude] = useState(null);

    useEffect(() => {
        setNavigation(props.navigation);
        setPicture(props.image);
        setLongitude(props.longitude);
        setLatitude(props.latitude);
    });

    return(
        <Modal 
        isVisible={props.visible} 
        onBackButtonPress={props.onCancel} 
        onBackdropPress={props.onCancel}
        animationOutTiming={3000}>
            <View style={styles.contentContainer}>
                <MapView 
                initialRegion={{
                    latitude: latitude, 
                    longitude: longitude,
                    latitudeDelta: 0.01,
                    longitudeDelta: 0.01,
                }}
                style={StyleSheet.absoluteFillObject}/>
            </View>
        </Modal>
    );
};

const styles = StyleSheet.create({
    contentContainer: {
        flex: 1,
        margin: 20,
        backgroundColor: "white",
        borderRadius: 10
    },
});

export default MapSignPicker;