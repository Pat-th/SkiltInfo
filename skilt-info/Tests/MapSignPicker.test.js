import React from "react";
import renderer from "react-test-renderer";

import MapSignPicker from "../components/MapSignPicker";

jest.mock('react-native-maps', () => {
    const { View } = require('react-native');
    const MockMapView = (props: any) => {
      return <View>{props.children}</View>;
    };
    const MockMarker = (props: any) => {
      return <View>{props.children}</View>;
    };
    return {
      __esModule: true,
      default: MockMapView,
      Marker: MockMarker,
    };
  });

describe("<MapSignPicker />", () => {
    it("Has 2 children", () => {
        const tree = renderer.create(<MapSignPicker />).toJSON();
        expect(tree.children.length).toBe(2);
    });
    it('Renders correctly', () => {
        const tree = renderer.create(<MapSignPicker />).toJSON();
        expect(tree).toMatchSnapshot();
      });
   });