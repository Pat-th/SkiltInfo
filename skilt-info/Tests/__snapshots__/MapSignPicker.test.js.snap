// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MapSignPicker /> Renders correctly 1`] = `
<Modal
  animationType="none"
  hardwareAccelerated={false}
  hideModalContentWhileAnimating={false}
  onModalHide={[Function]}
  onModalWillHide={[Function]}
  onModalWillShow={[Function]}
  onRequestClose={[Function]}
  scrollHorizontal={false}
  scrollOffset={0}
  scrollOffsetMax={0}
  scrollTo={null}
  supportedOrientations={
    Array [
      "portrait",
      "landscape",
    ]
  }
  swipeThreshold={100}
  transparent={true}
  visible={false}
>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "black",
        "bottom": 0,
        "height": 1334,
        "left": 0,
        "opacity": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
        "width": 750,
      }
    }
  />
  <View
    hideModalContentWhileAnimating={false}
    onModalHide={[Function]}
    onModalWillHide={[Function]}
    onModalWillShow={[Function]}
    pointerEvents="box-none"
    scrollHorizontal={false}
    scrollOffset={0}
    scrollOffsetMax={0}
    scrollTo={null}
    style={
      Object {
        "flex": 1,
        "justifyContent": "center",
        "margin": 37.5,
        "transform": Array [
          Object {
            "translateY": 0,
          },
        ],
      }
    }
    supportedOrientations={
      Array [
        "portrait",
        "landscape",
      ]
    }
    swipeThreshold={100}
  >
    <View
      style={
        Object {
          "backgroundColor": "white",
          "borderRadius": 10,
          "flex": 1,
          "margin": 20,
        }
      }
    >
      <View />
    </View>
  </View>
</Modal>
`;
